autoload/ale/lsp.vim

```diff
@@ -296,10 +299,19 @@ function! ale#lsp#HandleMessage(conn_id, message) abort
     " responses.
     if l:conn.initialized
         for l:response in l:response_list
-            " Call all of the registered handlers with the response.
-            for l:Callback in l:conn.callback_list
-                call ale#util#GetFunction(l:Callback)(a:conn_id, l:response)
-            endfor
+            if has_key(l:response, 'id') && has_key(s:custom_callbacks, l:response.id)
+                " Response to a custom request, call the registered one-shot handler.
+                try
+                    call s:custom_callbacks[l:response.id](l:response)
+                finally
+                    call remove(s:custom_callbacks, l:response.id)
+                endtry
+            else
+                " Call all of the registered handlers with the response.
+                for l:Callback in l:conn.callback_list
+                    call ale#util#GetFunction(l:Callback)(a:conn_id, l:response)
+                endfor
+            endif
```

Comment:
In order to avoid making this part of the code slower, let's keep this code how it is, and set up a callback which handles the message with the custom callbacks dictionary in the `lsp_linter.vim` file instead. That way we won't have to check for custom callbacks on every response when you never send custom requests.

Reply:
Done, I added a callback.